### PR TITLE
Feature/game lobby take 2

### PR DESCRIPTION
**Pull request by emilyseibert**
Issue #124 

* Implemented game loading separated by all open, my open games, all finished, and my finished games
* sort open type games by start time from past --> future to show ongoing games to future ongoing games
* sort finished games by end time from most recently ended --> ended earlier. 
* added Show My Games toggle with designs

### CODE REVIEW

**Comment by emilyseibert** [comment]
just realized this doesn't account for empty state so doing that now... hold off on reviewing

File: src/cljs/wombats_web_client/events/games.cljs [hunk 1]
```diff
@@ -26,33 +36,36 @@
...
 ;; TODO Scaling Issue with Lots of games - only update with games that are new?
 (defn get-open-games []
   (get-games
    "pending-open&status=pending-closed&status=active"
    #(re-frame/dispatch [:open-games %])
    #(print "error on get open games")))
 
+(defn get-my-open-games []
+  (get-my-games
+    "pending-open&status=pending-closed&status=active"
```
Comment (oconn) [inline src/cljs/wombats_web_client/events/games.cljs]: I would recommend pulling the first status into this query and then create a method that takes a set of statuses and builds the query. `(build-game-query #{:pending-open :pending-closed :active})` => `"status=pending-open&status=pending-closed&status=active"`

File: src/cljs/wombats_web_client/panels/games.cljs [hunk 2]
```diff
@@ -28,20 +27,46 @@
...
+  (let [current-state (:show-my-games @cmpnt-state)]
+    [:div.my-game-toggle-wrapper
+     [:div.checkbox {:class (when current-state "selected")
+                     :on-click #(swap! cmpnt-state assoc :show-my-games (not current-state))}] 
+     [:div.desc "SHOW MY GAMES"]]))
+
+(defn get-games [{:keys [show-open show-my-games open closed my-open my-closed]}]
+  (cond
+   (and show-open show-my-games) (sort-by :game/start-time my-open)
+   (and (not show-open) show-my-games) (reverse (sort-by :game/end-time my-closed))
```
Comment (oconn) [inline src/cljs/wombats_web_client/panels/games.cljs]: `sort-by` takes an optional argument that does the sorting for you. so you can probably do this;

```clj
(reverse (sort-by :game/end-time my-closed))
(sort-by :game/end-time < my-closed)
```

File: src/cljs/wombats_web_client/events/games.cljs [hunk 3]
```diff
@@ -26,33 +36,36 @@
...
    #(print "error on get all closed games")))
 
-(defn get-all-joined-games [user-id]
-  (get-joined-games
-   user-id
-   #(re-frame/dispatch [:joined-games %])
-   #(print "error with getting my games")))
+(defn get-my-closed-games []
+  (get-my-games
+   "closed"
```
Comment (dehli) [inline src/cljs/wombats_web_client/events/games.cljs]: Might be good to have all the statuses defined somewhere so they can easily be changed if need be.

File: src/cljs/wombats_web_client/panels/games.cljs [hunk 4]
```diff
@@ -41,27 +39,43 @@
...
-            show-open (:show-open @cmpnt-state)
-            games (get-games {:show-open show-open
-                              :show-my-games show-my-games
-                              :open open
-                              :closed closed
-                              :my-open @my-open
-                              :my-closed @my-closed})]
+      (let  [current-user @current-user
+             open @open-games
+                closed @closed-games
```
Comment (dehli) [inline src/cljs/wombats_web_client/panels/games.cljs]: spacing seems off here

Comment (emilyseibert) [inline src/cljs/wombats_web_client/panels/games.cljs]: that's because it is

File: src/cljs/wombats_web_client/events/games.cljs (hunk 1 above)
Comment (emilyseibert) [inline src/cljs/wombats_web_client/events/games.cljs]: Issue #146

**Pull request merged**